integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/common/EndPointParser.java

```diff
@@ -11,20 +13,37 @@ public class EndPointParser {
     private EndPointParser() {
     }
 
-    public static Optional<String> getEndpointBaseUrl(String endpoint) {
+    public static Optional<String> getBaseUrl(String endpoint) {
         Matcher matcher = ENDPOINT_URL_REGEX.matcher(endpoint);
         if (matcher.find()) {
             return Optional.of(matcher.group(1));
         }
         return Optional.empty();
     }
 
-    public static Optional<String> getEndpointPathUrl(String endpoint) {
+    public static Optional<String> getPathUrl(String endpoint) {
         Matcher matcher = ENDPOINT_URL_REGEX.matcher(endpoint);
         if (matcher.find()) {
             return Optional.of(matcher.group(2));
         }
         return Optional.empty();
     }
 
+    public static Optional<String> getAuthority(String endpoint) {
+        try {
+            return Optional.of(new URL(endpoint).getAuthority());
+        } catch (MalformedURLException e) {
+            return Optional.empty();
+        }
+    }
+
+    public static Optional<Integer> getEndpointPort(String endpoint) {
+        try {
+            int port = new URL(endpoint).getPort();
+            return Optional.of(port != -1 ? port : 80);
```

Comment:
Can you please add also handling of default HTTPS port?

Reply:
method not needed, removed.